api/filters/platform_termination_filter.go

```diff
@@ -62,11 +65,68 @@ func (f *platformTerminationFilter) Run(req *web.Request, next web.Handler) (*we
 				StatusCode:  http.StatusUnprocessableEntity,
 			}
 		}
+		sharingReferences, err := findSharingReferenceInstancesFromOtherPlatform(ctx, platform, f.repository)
+		if err != nil {
+			return nil, err
+		}
+		if len(sharingReferences) > 0 {
+			return nil, &util.HTTPError{
+				ErrorType:   "UnprocessableEntity",
+				Description: "Platform cannot be deleted because other platform(s) has reference instance(s) to the shared instances in given platform. Details: " + formatSharingReferences(sharingReferences),
+				StatusCode:  http.StatusUnprocessableEntity,
+			}
+		}
 	}
-
 	return next.Handle(req)
 }
 
+func formatSharingReferences(references []*SharingReferences) string {
+	var msg []string
+	for i := 0; i < len(references); i++ {
+		msg = append(msg, "shared instance "+references[i].sharedInstanceID+" referenced by instance(s) "+strings.Join(references[i].referenceIDs, ", "))
+	}
+	return strings.Join(msg, ", ")
+}
+
+type SharingReferences struct {
+	sharedInstanceID string
+	referenceIDs     []string
+}
+
+func findSharingReferenceInstancesFromOtherPlatform(ctx context.Context, platform *types.Platform, repository storage.Repository) ([]*SharingReferences, error) {
```

Comment:
consider changing this method name to findSharedInstancesRefferncesInOtherPlatforms